packages/@aws-cdk/aws-codepipeline-actions/lib/bitbucket/source-action.ts

```diff
@@ -0,0 +1,107 @@
+import * as codepipeline from '@aws-cdk/aws-codepipeline';
+import * as iam from '@aws-cdk/aws-iam';
+import { Construct } from '@aws-cdk/core';
+import { Action } from '../action';
+import { sourceArtifactBounds } from '../common';
+
+/**
+ * Construction properties for {@link BitBucketSourceAction}.
+ *
+ * @experimental
+ */
+export interface BitBucketSourceActionProps extends codepipeline.CommonAwsActionProps {
+  /**
+   * The output artifact that this action produces.
+   * Can be used as input for further pipeline actions.
+   */
+  readonly output: codepipeline.Artifact;
+
+  /**
+   * The ARN of the CodeStar Connection created in the AWS console
+   * that has permissions to access this BitBucket repository.
+   *
+   * @example 'arn:aws:codestar-connections:us-east-1:123456789012:connection/12345678-abcd-12ab-34cdef5678gh'
+   */
+  readonly connection: string;
+
+  /**
+   * The owning user or organization of the repository.
+   *
+   * @example 'aws'
+   */
+  readonly owner: string;
+
+  /**
+   * The name of the repository.
+   *
+   * @example 'aws-cdk'
+   */
+  readonly repo: string;
+
+  /**
+   * The branch to build.
+   *
+   * @default 'master'
+   */
+  readonly branch?: string;
+
+  /**
+   * Whether the output should be the contents of the repository
+   * (which is the default),
+   * or a link that allows CodeBuild to clone the repository before building.
+   *
```

Comment:
Is there any AWS docs reference that explains this feature. I couldn't understand what this does or what is the use case that this addresses from the description here...

Reply:
I don't fully understand it either, honestly, and I couldn't get it to work. Will add a link.